idd/schema/test_idd_parser.py

```diff
@@ -0,0 +1,169 @@
+import pytest
+import idd_parser
+
+
+# If this timeouts while you make modifications, you can do `pip install pytest-timeout` and use this mark here:
+# @pytest.mark.timeout(2)
+def test_ok_idd():
+    data = idd_parser.Data()
+    data.file = r"""
+! Comment
+
+\group test
+
+OkObject,
+  A1; \field Name
+      \minimum 1
+
+NotBrokenObject,
+  A1, \field Name
+  N2; \field Last Field
+      \maximum 10
+
+Obj2,
+  A1; \field Name
+"""
+    idd_parser.parse_idd(data)
+    assert data.schema["properties"].keys() == {"OkObject", "NotBrokenObject", "Obj2"}
+    assert data.schema["properties"]["NotBrokenObject"] == {
+        "patternProperties": {
+            ".*": {"type": "object", "properties": {"last_field": {"type": "number", "maximum": 10.0}}}
+        },
+        "group": "test",
+        "name": {"type": "string"},
+        "legacy_idd": {
+            "field_info": {
+                "name": {"field_name": "Name", "field_type": "a"},
+                "last_field": {"field_name": "Last Field", "field_type": "n"},
+            },
+            "fields": ["name", "last_field"],
+            "alphas": {"fields": ["name"]},
+            "numerics": {"fields": ["last_field"]},
+        },
+        "type": "object",
+    }
+
+
+def test_broken_idd():
+    data = idd_parser.Data()
+    data.file = r"""! Comment
+
+\group test
+
+OkObject,
+  A1; \field Name
+      \minimum 1
+
+BrokenObject,
+  A1, \field Name
+  A2, \field Last Field
+      \maximum 10
+
+Obj2,
+  A1; \field Name
+"""
+    with pytest.raises(idd_parser.MissingSemiColonException) as e:
+        idd_parser.parse_idd(data)
+
+    assert (
+        str(e.value)
+        == r"""In object 'BrokenObject', Missing semi-colon in field A2
+Context:
+  A1; \field Name
+      \minimum 1
+
+BrokenObject,
+  A1, \field Name
+  A2, \field Last Field
+    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      \maximum 10
+
+Obj2,
+  A1; \field Name"""
+    )
```

Comment:
New test file that uses pytest (more on this right after). Here's the Missing Semi-Colon test for eg.